terraform/ecs_asg/ecs_asg_launch_config/security_groups.tf

```diff
@@ -1,7 +1,7 @@
 resource "aws_security_group" "instance_sg" {
   description = "controls direct access to application instances"
   vpc_id      = "${var.vpc_id}"
-  name        = "${var.asg_name}_instance_sg"
+  name        = "${var.asg_name}_instance_sg_${random_id.sg_append.hex}"
```

Comment:
The random ID along with `create_before_destroy` avoids a dependency issue where the old SGs cannot be deleted as they are in use.